Makefile

```diff
@@ -136,6 +137,17 @@ $(TEST): $(SOURCE) $(SPECS) $(BIN)
 testem:
 	$(BIN)/testem
 
+# Rules to attach checksums to JavaScript source in the header
+app.html: app.html.in .make-script-block
+	sed -e '/%SCRIPTS%/ {' -e 'r .make-script-block' -e 'd' -e '}' app.html.in > app.html
+
+.make-script-block: $(SOURCE)
+	> .make-script-block
+	for file in $(SOURCE); do \
+		checksum=$$(md5 -q $$file) ; \
```

Comment:
But that would only work on OS X (and other Unix variants) but not Linux (which doesn't have md5sum). You'll want to conditionally determine what is available and then use that conditional. This test seems to work:

```Makefile
# Detemine MD5 utility
ifeq ($(shell which md5),)
    MD5 = md5sum
else 
    MD5 = md5 -q
endif

md5:
    echo $(MD5)
```

That `md5` target is just there for demonstration, but not to be included in the real makefile. Then the command above will change to:

```Makefile
checksum=$$($(MD5) $$file | awk '{ print $$1 }') ; \
```

And you'll need that `awk` command in there which will work with the output of either format of the MD5 utility, which was my real motive in suggesting the change before.